android/src/main/kotlin/eu/nk2/tinkoff_acquiring_sdk/TinkoffProtocol.kt

```diff
@@ -22,24 +23,78 @@ data class TinkoffOrderOptions(
     val money: Money,
     val title: String,
     val description: String,
-    val recurrentPayment: Boolean
+    val recurrentPayment: Boolean,
+    val shops: List<TinkoffShop>?,
+    val receipt: TinkoffReceipt?
+)
+
+data class TinkoffShop(
+    val shopCode: String? = null,
+    val name: String? = null,
+    val amount: Long? = 0,
+    val fee: String? = null
+)
+
+data class TinkoffReceipt(
+    val shopCode: String? = null,
+    val email: String? = null,
+    val taxation: String? = null,
+    val phone: String? = null,
+    val items: List<TinkoffItem>? = null
+)
+
+data class TinkoffItem(
+    var price: Long = 0,
+    var quantity: Double = 0.0,
+    var name: String = "",
+    var amount: Long = 0,
+    var tax: Tax? = null
 )
 
 fun MethodCall.toTinkoffOrderOptions() = TinkoffOrderOptions(
     orderId = this.argument("orderId") ?: error("orderId is required in openPaymentScreen method"),
     title = this.argument("title") ?: error("title is required in openPaymentScreen method"),
     description = this.argument("description") ?: error("description is required in openPaymentScreen method"),
     money = Money.ofRubles(this.argument<Double>("money") ?: error("money is required in openPaymentScreen method")),
-    recurrentPayment = this.argument("recurrentPayment") ?: false
+    recurrentPayment = this.argument("recurrentPayment") ?: false,
+    shops = this.argument<List<Map<String, Any>>>("shops")?.map {
+        TinkoffShop(
+            it["shopCode"]?.toString(),
```

Comment:
elvis with IllegalStateException should be used to enforce non-nullable parameters
assert checks should be done on frontend (dart)